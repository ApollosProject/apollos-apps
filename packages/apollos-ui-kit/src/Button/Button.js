import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import { compose } from 'recompose';
import { get } from 'lodash';

import { withTheme, withThemeMixin } from '../theme';
import styled from '../styled';
import Touchable from '../Touchable';
import { H4 } from '../typography';
import { InlineActivityIndicator } from '../ActivityIndicator';
import { withPlaceholder, Line } from '../Placeholder';

const ButtonStyles = styled(
  ({ theme, disabled, bordered, pill }) => ({
    paddingHorizontal: theme.sizing.baseUnit,
    overflow: 'hidden',
    borderRadius: pill ? theme.sizing.baseUnit * 3 : theme.sizing.baseUnit / 2,
    flexDirection: 'row',
    height: theme.sizing.baseUnit * 3,
    alignItems: 'center',
    justifyContent: 'center',
    opacity: disabled ? 0.5 : 1,
    borderWidth: 2,
    ...(bordered
      ? {
          backgroundColor: theme.colors.transparent,
          borderColor: theme.colors.primary,
        }
      : {
          backgroundColor: theme.colors.background.screen,
          borderColor: theme.colors.background.screen,
          elevation: 2,
        }),
  }),
  'Button'
)(View);

const ButtonPlaceholder = styled(
  ({ theme }) => ({
    width: theme.sizing.baseUnit * 4,
    height: theme.sizing.baseUnit + theme.helpers.rem(1),
  }),
  'Button.Placeholder'
)(Line);

export const withButtonPlaceholder = withPlaceholder(ButtonPlaceholder);

const enhance = compose(
  withButtonPlaceholder,
  withTheme(({ theme, type = 'default', pill }) => ({
    fill: get(theme, `buttons.${type}.fill`, theme.colors.action.default),
    accent: get(theme, `buttons.${type}.accent`, theme.colors.text.primary),
    borderRadius: pill ? theme.sizing.baseUnit * 3 : theme.sizing.baseUnit / 2,
  })),
  // makes non-text children inherit button styles by default :-D
  withThemeMixin(({ fill, accent, bordered }) => {
    const textColor = bordered ? fill : accent;
    return {
      colors: {
        primary: bordered ? fill : accent,
        text: {
          primary: textColor,
          secondary: textColor,
          tertiary: textColor,
        },
        screen: fill,
      },
    };
  }),
  withTheme()
);

// API-Compatible to React-Native's base <Button> component,
// except it supports the rendering of children, which will take precedence over the title prop,
// so you can handle non text children.
const Button = enhance(
  ({
    children,
    disabled,
    title,
    to,
    onPress,
    style,
    bordered,
    loading,
    accent,
    pill,
    TouchableComponent = Touchable,
    theme,
    ...touchableProps
  }) => {
    const accessibilityTraits = ['button'];
    if (disabled || loading) accessibilityTraits.push('disabled');

    const buttonContent = (
      <ButtonStyles
        style={style}
        disabled={disabled}
        bordered={bordered}
        pill={pill}
      >
        {loading ? (
          <InlineActivityIndicator color={accent} />
        ) : (
          children || <H4>{title}</H4>
        )}
      </ButtonStyles>
    );

    if (onPress) {
      return (
        <TouchableComponent
          onPress={onPress}
          disabled={disabled || loading}
          accessibilityTraits={accessibilityTraits}
          {...touchableProps}
        >
          {buttonContent}
        </TouchableComponent>
      );
    }

    return buttonContent;
  }
);

Button.defaultProps = {
  disabled: false,
  bordered: false,
  pill: true,
  title: '',
  accessibilityComponentType: 'button',
  type: 'primary',
};

Button.propTypes = {
  disabled: PropTypes.bool,
  children: PropTypes.node,
  title: PropTypes.string,
  bordered: PropTypes.bool,
  pill: PropTypes.bool,
  to: PropTypes.string,
  type: PropTypes.oneOf([
    'default',
    'primary',
    'secondary',
    'tertiary',
    'ghost',
    'alert',
  ]), // keys in theme.colors.action
  ...Touchable.propTypes,
};

export default Button;
