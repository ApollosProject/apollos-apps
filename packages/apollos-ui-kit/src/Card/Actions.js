import PaddedView from '../PaddedView';
import styled from '../styled';

const Actions = styled(
  ({ theme }) => ({
    flexWrap: 'wrap',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingTop: theme.sizing.baseUnit / 4,
  }),
  'Card.Actions'
)(PaddedView);

export default Actions;
