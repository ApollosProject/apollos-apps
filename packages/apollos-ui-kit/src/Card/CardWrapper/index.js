import { compose, pure } from 'recompose';
import React from 'react';
import { Platform, View } from 'react-native';
import PropTypes from 'prop-types';

import { withIsLoading } from '../../isLoading';
import styled from '../../styled';

const StyledCard = compose(
  withIsLoading,
  styled(({ theme, cardColor, inHorizontalList = false }) => ({
    // card styles
    backgroundColor: cardColor || theme.colors.background.paper || undefined, // bail out if no bg color
    borderRadius: theme.sizing.baseUnit,
    ...(inHorizontalList
      ? {
          // provides spacing between cards also fixes android shadow needing "space" to render into
          margin: theme.sizing.baseUnit / 2,
        }
      : {
          marginHorizontal: theme.sizing.baseUnit,
          marginVertical: theme.sizing.baseUnit * 0.75,
        }),
    ...Platform.select(theme.shadows.default),
  }))
)(View);

/*
 * Overflow on iOS, when declared on the same element as a shadow, clips the shadow so overflow must
 * live on a child wrapper. https://github.com/facebook/react-native/issues/449
 */
const OverflowFix = styled(({ theme, forceRatio }) => ({
  borderRadius: theme.sizing.baseUnit,
  overflow: 'hidden',
  aspectRatio: forceRatio,
}))(View);

const Card = pure(({ children, isLoading, forceRatio, ...otherProps }) => (
  <StyledCard forceRatio={forceRatio} {...otherProps}>
    <OverflowFix forceRatio={forceRatio}>{children}</OverflowFix>
  </StyledCard>
));

Card.propTypes = {
  backgroundColor: PropTypes.string,
  children: PropTypes.node,
  forceRatio: PropTypes.number,
  style: PropTypes.any, // eslint-disable-line
};

export default Card;
