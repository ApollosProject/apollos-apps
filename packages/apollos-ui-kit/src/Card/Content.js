import PaddedView from '../PaddedView';
import styled from '../styled';

const Content = styled(
  ({ theme }) => ({
    paddingVertical: theme.sizing.baseUnit,
  }),
  'Card.Content'
)(PaddedView);

export default Content;
