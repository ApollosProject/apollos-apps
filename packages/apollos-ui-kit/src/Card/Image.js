import { Platform } from 'react-native';
import { compose } from 'recompose';

import GradientOverlayImage from '../GradientOverlayImage';
import styled from '../styled';
import { getIsLoading } from '../isLoading';

const Image = compose(
  getIsLoading,
  styled(
    ({ theme }) => ({
      width: '100%',
      ...Platform.select({
        android: {
          // fixes android borderRadius overflow display issue
          borderTopRightRadius: theme.sizing.baseUnit,
          borderTopLeftRadius: theme.sizing.baseUnit,
        },
      }),
    }),
    'Card.Image'
  )
)(GradientOverlayImage);

Image.propTypes = GradientOverlayImage.propTypes;

export default Image;
