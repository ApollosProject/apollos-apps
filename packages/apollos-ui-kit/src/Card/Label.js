import React from 'react';
import { get } from 'lodash';
import PropTypes from 'prop-types';

import styled from '../styled';
import { ThemeMixin } from '../theme';
import Chip from '../Chip';
import Icon from '../Icon';

const StyledChip = styled(
  ({ type }) => ({
    /* when the overlay type is transparent the borders look funny. TODO: Button refactor - consider
     * moving this change to `Chip` or `Button` */
    ...(type === 'overlay' ? { borderWidth: 0 } : {}),
  }),
  'ui-kit.CardLabel'
)(Chip);

const Label = ({ title, icon, type, theme, IconComponent, ...props }) => (
  <ThemeMixin
    mixin={{
      type: get(theme, 'type', 'light').toLowerCase(), // not sure why we need toLowerCase
      colors: get(theme, 'colors', {}),
    }}
  >
    <StyledChip title={title} type={type} icon={icon} {...props}>
      {React.isValidElement(IconComponent) || !IconComponent ? (
        IconComponent
      ) : (
        <IconComponent />
      )}
    </StyledChip>
  </ThemeMixin>
);

Label.propTypes = {
  title: PropTypes.string.isRequired,
  icon: Icon.propTypes.name,
  theme: PropTypes.shape({
    type: PropTypes.string,
    colors: PropTypes.shape({}),
  }),
  type: PropTypes.string,
  IconComponent: PropTypes.oneOfType([
    PropTypes.elementType,
    PropTypes.element,
  ]),
  ...Chip.propTypes,
};

Label.defaultProps = {
  type: 'overlay',
};

Label.displayName = 'CardLabel';

export default Label;
