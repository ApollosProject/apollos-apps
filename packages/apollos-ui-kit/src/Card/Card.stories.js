import React from 'react';
import { storiesOf } from '@storybook/react-native';

import { H3, H6, BodyText, Paragraph } from '../typography';
import Button, { ButtonLink } from '../Button';

import Card, { CardActions, CardContent, CardImage } from '.';

storiesOf('Card', module)
  .add('simple', () => (
    <Card>
      <CardImage source={'https://picsum.photos/600/400/?image=63'} />
      <CardContent>
        <H3>Coffee</H3>
        <H6 padded>noun</H6>
        <Paragraph>
          <BodyText>
            {
              'A dark substance that turns "leave me alone" into "good morning!"'
            }
          </BodyText>
        </Paragraph>
      </CardContent>
      <CardActions>
        <Button title="Learn More" pill={false} />
        <ButtonLink>Share</ButtonLink>
      </CardActions>
    </Card>
  ))
  .add('loading', () => (
    <Card isLoading>
      <CardImage />
      <CardContent>
        <H3 />
        <H6 padded />
        <Paragraph>
          <BodyText />
        </Paragraph>
      </CardContent>
      <CardActions>
        <Button pill={false} />
        <ButtonLink />
      </CardActions>
    </Card>
  ));
