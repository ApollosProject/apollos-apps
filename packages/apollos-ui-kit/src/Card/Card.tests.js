import React from 'react';
import renderer from 'react-test-renderer';

import Providers from '../Providers';
import { H3, H6, BodyText, Paragraph } from '../typography';
import Button, { ButtonLink } from '../Button';

import Card, { CardImage, CardContent, CardActions } from '.';

describe('the Card component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <Card>
          <CardImage source={'https://picsum.photos/600/400/?image=63'} />
          <CardContent>
            <H3>Coffee</H3>
            <H6 padded>noun</H6>
            <Paragraph>
              <BodyText>
                {
                  'A dark substance that turns "leave me alone" into "good morning!"'
                }
              </BodyText>
            </Paragraph>
          </CardContent>
          <CardActions>
            <Button title="Learn More" pill={false} />
            <ButtonLink>Share</ButtonLink>
          </CardActions>
        </Card>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('it should render a placeholder', () => {
    const tree = renderer.create(
      <Providers>
        <Card isLoading>
          <CardContent>
            <H3 />
            <H6 padded />
            <Paragraph>
              <BodyText />
            </Paragraph>
          </CardContent>
          <CardActions>
            <Button pill={false} />
            <ButtonLink />
          </CardActions>
        </Card>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
