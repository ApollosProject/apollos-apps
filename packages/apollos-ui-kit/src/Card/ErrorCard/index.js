import React from 'react';
import PropTypes from 'prop-types';

import styled from '../../styled';
import { H4, H6 } from '../../typography';
import Card, { CardContent } from '..';

const Header = styled({ textAlign: 'center' }, 'Error.Header')(H4);
const ErrorText = styled({ textAlign: 'center' }, 'Error.Text')(H6);

const Err = ({ message = 'Uh oh!', error }) => {
  let errorMessage;
  if (typeof error !== 'string') {
    if (error && error.message) {
      errorMessage = error.message;
    } else if (error && error.error && typeof error.error === 'string') {
      errorMessage = error.error;
    } else {
      errorMessage = 'An unexpected error occured';
    }
  } else {
    errorMessage = error;
  }

  return (
    <Card>
      <CardContent>
        <Header>{message}</Header>
        <ErrorText>{errorMessage}</ErrorText>
      </CardContent>
    </Card>
  );
};

Err.propTypes = {
  message: PropTypes.string,
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

export default Err;
