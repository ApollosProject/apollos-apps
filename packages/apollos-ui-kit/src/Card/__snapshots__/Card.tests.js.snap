// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Card component it should render a placeholder 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 16,
          "marginHorizontal": 16,
          "marginVertical": 12,
          "shadowColor": "rgba(165, 165, 165, 0.30000000000000004)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": undefined,
            "borderRadius": 16,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 24,
                "lineHeight": 27.6,
              }
            }
          />
          <Text
            padded={true}
            style={
              Object {
                "color": "#505050",
                "fontFamily": "InterUI-Bold",
                "fontSize": 12,
                "lineHeight": 17.28,
                "paddingBottom": 8.64,
              }
            }
          />
          <View
            style={
              Object {
                "paddingBottom": 17.28,
              }
            }
          >
            <Text
              bold={false}
              italic={false}
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "InterUI-Regular",
                  "fontSize": 16,
                  "lineHeight": 26,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
              "paddingHorizontal": 16,
              "paddingTop": 4,
              "paddingVertical": 16,
            }
          }
        >
          <View
            bordered={false}
            disabled={false}
            pill={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00676D",
                "borderColor": "#00676D",
                "borderRadius": 8,
                "borderWidth": 2,
                "elevation": 2,
                "flexDirection": "row",
                "height": 48,
                "justifyContent": "center",
                "opacity": 1,
                "overflow": "hidden",
                "paddingHorizontal": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "InterUI-Bold",
                  "fontSize": 16,
                  "lineHeight": 23.04,
                }
              }
            >
              
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "#17B582",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`the Card component should render 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 16,
          "marginHorizontal": 16,
          "marginVertical": 12,
          "shadowColor": "rgba(165, 165, 165, 0.30000000000000004)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": undefined,
            "borderRadius": 16,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#A5A5A5",
              "overflow": "hidden",
              "width": "100%",
            }
          }
        >
          <View
            animate={null}
            customAnimate={null}
            onReady={false}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#A5A5A5",
                "width": "100%",
              }
            }
          >
            <Image
              onLoad={[Function]}
              source={
                Array [
                  Object {
                    "cache": "force-cache",
                    "uri": "https://picsum.photos/600/400/?image=63",
                    "url": "https://picsum.photos/600/400/?image=63",
                  },
                ]
              }
              style={
                Object {
                  "backgroundColor": "#A5A5A5",
                  "opacity": 0,
                  "resizeMode": "cover",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 24,
                "lineHeight": 27.6,
              }
            }
          >
            Coffee
          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#505050",
                "fontFamily": "InterUI-Bold",
                "fontSize": 12,
                "lineHeight": 17.28,
                "paddingBottom": 8.64,
              }
            }
          >
            noun
          </Text>
          <View
            style={
              Object {
                "paddingBottom": 17.28,
              }
            }
          >
            <Text
              bold={false}
              italic={false}
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "InterUI-Regular",
                  "fontSize": 16,
                  "lineHeight": 26,
                }
              }
            >
              A dark substance that turns "leave me alone" into "good morning!"
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
              "paddingHorizontal": 16,
              "paddingTop": 4,
              "paddingVertical": 16,
            }
          }
        >
          <View
            bordered={false}
            disabled={false}
            pill={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00676D",
                "borderColor": "#00676D",
                "borderRadius": 8,
                "borderWidth": 2,
                "elevation": 2,
                "flexDirection": "row",
                "height": 48,
                "justifyContent": "center",
                "opacity": 1,
                "overflow": "hidden",
                "paddingHorizontal": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "InterUI-Bold",
                  "fontSize": 16,
                  "lineHeight": 23.04,
                }
              }
            >
              Learn More
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "#17B582",
              }
            }
          >
            Share
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
